### PR TITLE
Biospecimen Dashboard: Added API endpoints to query participants for Daily Reports Page

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
Title^^
This PR addresses following issue:
https://github.com/episphere/connect/issues/631
Related PR:
https://github.com/episphere/biospecimen/pull/569

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -2125,6 +2125,53 @@ const updateUserPhoneSigninMethod = async (phone, uid) => {
...
+ * returns participants that are no more than 2 days old from the time of check in date/time
+ * Using promises to resolve array of objects
+ * object containes essential information Collection Location (951355211), Connect ID,
+ * F+L Name (996038075 & 399159511), Check-In (840048338), Collection ID (820476880),
+ * Collection Finalized(556788178), Check-Out (343048998)
+ * Queried from Participants & Biospecimen table
+ */
+
+const queryDailyReportParticipants = async () => {
+    const twoDaysAgo = new Date(new Date().getTime() - (2 * 24 * 60 * 60 * 1000)).toISOString();
```
Comment (JoeArmani) [inline utils/firestore.js:2140]: Comment only:
This is great for a rolling 48-hour window. Might be worth hardcoding the calculation (micro-optimization).
If, in the future, there's a preference for calendar days as the reporting window, something like this should work:
```javascript
const twoDaysAgo = new Date();
twoDaysAgo.setDate(twoDaysAgo.getDate() - 2);
twoDaysAgo.setHours(0, 0, 0, 0);
const twoDaysAgoIso = twoDaysAgo.toISOString();
```

File: utils/firestore.js [hunk 2]
```diff
@@ -2125,6 +2125,53 @@ const updateUserPhoneSigninMethod = async (phone, uid) => {
...
+    let query = db.collection('participants');
+    const snapshot = await query.where('331584571.266600170.135591601', '==', 353358909).where('331584571.266600170.840048338', '<', twoDaysAgo).get();
+    if (snapshot.size !== 0) {
+        const promises = snapshot.docs.map(async (document) => {
+            return processQueryDailyReportParticipants(document);
+        });
+        return Promise.all(promises).then((results) => {
+            return results.filter((result) => result !== undefined);
```
Comment (JoeArmani) [inline utils/firestore.js:2148]: Question: Do we have to worry about null or falsy values here?
`return results.filter((result) => result != null);` //filters null and undefined
`return results.filter((result) => result);` //filters anything falsy

File: utils/firestore.js [hunk 3]
```diff
@@ -2125,6 +2125,53 @@ const updateUserPhoneSigninMethod = async (phone, uid) => {
...
+            dailyReport['951355211'] = document.data()['951355211'];
+            dailyReport['343048998'] = document.data()['331584571']['266600170']?.['343048998'];
+            dailyReport['951355211'] = secondSnapshot.docs[0].data()['951355211'];
+            dailyReport['820476880'] = secondSnapshot.docs[0].data()['820476880'];
+            dailyReport['556788178'] = secondSnapshot.docs[0].data()['556788178'];
+            
+            return dailyReport;
+        }
+    }
+};
```
Comment (JoeArmani) [inline utils/firestore.js:2173]: Please add try/catch.
Please consider returning an explicit empty value when `secondSnapshot.size === 0` to handle the unexpected empty case in a specific way.

**Review by JoeArmani — CHANGES_REQUESTED** [review]

File: utils/firestore.js (hunk 2 above)
Comment (abhinavjonnada82) [reply utils/firestore.js:2148]: i dont think so

File: utils/firestore.js [hunk 4]
```diff
@@ -2125,6 +2125,53 @@ const updateUserPhoneSigninMethod = async (phone, uid) => {
...
+        const promises = snapshot.docs.map(async (document) => {
+            return processQueryDailyReportParticipants(document);
+        });
+        return Promise.all(promises).then((results) => {
+            return results.filter((result) => result !== undefined);
+        });
+    } else {
+        return 'Error';
+    }
+};
```
[the remark this answers is not in the document]
Comment (abhinavjonnada82) [reply utils/firestore.js:2153]: my bad I normally do add try/catch blocks & as I did in my other code.
I'm fully aware of the side effects

**Review by JoeArmani — APPROVED** [review]
looks good

**Review by anthonypetersen — APPROVED** [review]
looks good

**Pull request merged**